lib/tls.js

```diff
@@ -60,6 +63,10 @@ function convertProtocols(protocols) {
   const lens = new Array(protocols.length);
   const buff = Buffer.allocUnsafe(protocols.reduce((p, c, i) => {
     var len = Buffer.byteLength(c);
+    if (len > 255) {
+      throw new ERR_OUT_OF_RANGE('', '<= 255', len, 'The byte length of the ' +
+        `protocol at index ${i} exceeds the maximum length.`);
```

Comment:
When looking at this signature it looks like a mistake that the first argument is an empty string.

I suggest to use the following instead:

```js
ERR_OUT_OF_RANGE(str, range, input, replaceDefaultBoolean)
```

Using booleans as arguments is not great either but that way the first argument is at least used properly and the reader will hopefully be less surprised (as I would be). We might also use a different more expressive value than a boolean but that could be more error prone.

Reply:
I've refactored as per your suggestion. Check it out.